ui/app/adapters/generated-item-list.js

```diff
@@ -6,42 +6,111 @@
 import ApplicationAdapter from './application';
 import { task } from 'ember-concurrency';
 import { service } from '@ember/service';
+import { sanitizePath } from 'core/utils/sanitize-path';
+import { encodePath } from 'vault/utils/path-encoding-helpers';
+import { tracked } from '@glimmer/tracking';
 
-export default ApplicationAdapter.extend({
-  store: service(),
-  namespace: 'v1',
-  urlForItem() {},
-  dynamicApiPath: '',
+export default class GeneratedItemListAdapter extends ApplicationAdapter {
+  @service store;
+  namespace = 'v1';
 
-  getDynamicApiPath: task(function* (id) {
-    // TODO: remove yield at some point.
-    const result = yield this.store.peekRecord('auth-method', id);
-    this.dynamicApiPath = result.apiPath;
-    return;
-  }),
+  // these items are set within getNewAdapter in path-help service
+  @tracked apiPath = '';
+  paths = {};
 
-  fetchByQuery: task(function* (store, query, isList) {
+  // These are the paths used for the adapter actions
+  get getPath() {
+    return this.paths.getPath || '';
+  }
+  get createPath() {
+    return this.paths.createPath || '';
+  }
+  get deletePath() {
+    return this.paths.deletePath || '';
+  }
+
+  getDynamicApiPath(id) {
+    const result = this.store.peekRecord('auth-method', id);
+    this.apiPath = result.apiPath;
+    return result.apiPath;
+  }
+
+  fetchByQuery = task(async (store, query, isList) => {
```

Comment:
Curious why this is a concurrency task? I don't see us calling any of the normal concurrency properties/states such as `isRunning` etc. Would making it an async/await suffice?